src/features/script-load.js

```diff
@@ -14,9 +14,47 @@ systemJSPrototype.createScript = function (url) {
   return script;
 };
 
+// Auto imports -> script tags can be inlined directly for load phase
+var lastAutoImportUrl, lastAutoImportDeps;
+var autoImportCandidates = {};
+var systemRegister = systemJSPrototype.register;
+var timeoutCnt = 0;
+systemJSPrototype.register = function (deps, declare) {
+  if (hasDocument && document.readyState === 'loading' && typeof deps !== 'string') {
+    var scripts = document.getElementsByTagName('script');
+    var lastScript = scripts[scripts.length - 1];
+    var url = lastScript && lastScript.src;
+    if (url) {
+      lastAutoImportUrl = url;
+      lastAutoImportDeps = deps;
+      autoImportCandidates[url] = [deps, declare];
+      var loader = this;
+      // This timeout ensures that if this is a dynamic script injection by SystemJS
```

Comment:
Does the event loop guarantee that dynamically injected scripts won't be executed in the same tick of the event loop. I'm thinking specifically when the file is cached, so there is no waiting on network.

Reply:
They don't, but even if they did it wouldn't matter as the important timing is the sync load event on the script itself.